### PR TITLE
Test doxy layout

### PR DESCRIPTION
**Pull request by m-kuhn**
## Description

More documentation safety checks

### CODE REVIEW

File: src/analysis/processing/qgsalgorithmtinmeshcreation.cpp [hunk 1]
```diff
@@ -132,17 +134,13 @@ QVariantMap QgsTinMeshCreationAlgorithm::processAlgorithm( const QVariantMap &pa
     destinationCrs = context.project()->crs();
   triangulation.setCrs( destinationCrs );
 
-  if ( !mVerticesLayer.isEmpty() && feedback )
```
Comment (nyalldawson) [inline src/analysis/processing/qgsalgorithmtinmeshcreation.cpp:135]: What happened here? Bad rebase?

Comment (m-kuhn) [reply src/analysis/processing/qgsalgorithmtinmeshcreation.cpp:135]: ouch

**Comment by m-kuhn** [comment]
I can't fix

```
/home/runner/work/QGIS/QGIS/src/core/labeling/qgspallabeling.h:272: error: documented symbol `enum Q_DECL_DEPRECATED QgsPalLayerSettings::QgsPalLayerSettings::LinePlacementFlags' was not declared or defined. (warning treated as error, aborting now)
```

What I tried is to define `Q_DECL_DEPRECATED=`. This works but triggers a new set of issues for the check that `\deprecated` is also `Q_DECL_DEPRECATED`. I tried aliasing `Q_DECL_DEPRECATED_ENUM` to `Q_DECL_DEPRECATED` and defining this to `Q_DECL_DEPRECATED_ENUM=`. This works but would require a myriad of sipify patches.
@nyalldawson ok to just `@cond PRIVATE` the two affected enums? Or drop the `Q_DECL_DEPRECATED` and go for dox only?

**Comment by nyalldawson** [comment]
> Or drop the Q_DECL_DEPRECATED and go for dox only?

Yep, try this one first?

**Pull request merged**